specs/contract_signatures/01_main.rst

```diff
@@ -0,0 +1,61 @@
+==========================================================
+Contract Signatures
+==========================================================
+
+=====  ===================
+epic      title       
+=====  ===================
+`78`_  Contract Signatures
+=====  ===================
+
+.. _78: https://github.com/gnosis/safe/issues/78
+
+.. _Main:
+
+
+#. `Main`_
+
+.. sectnum::
+.. contents:: Table of Contents
+    :local:
+    :depth: 2
+
+Problem Definition
+---------------------
+
+
+Technical Details
+-----------------
+
+Diagram_
+
+1. Dapp sends a JSON payload request to the browser extension.
+
+2. The chome extension signs the hash of the payload according to EIP-712_ and sends the following ``message`` to the app (via push notification) (See RootMessage_ for the root object used with the Notification Service).
+
+.. code:: javascript
+    
+    {
+        "type": "signTypedData"
+        "payload": <string> // Payload to sign
+        "hash": <hex-string> // Signature of the hash of the payload according to EIP712

```

Comment:
Since this serves more as request for confirmations then I would say that we can remove the `hash` here. This same payload can be used by the app to request confirmations from other signers.